packages/next/server/web/adapter.ts

```diff
@@ -162,6 +162,50 @@ export function blockUnallowedResponse(
   })
 }
 
+export function enhanceGlobals() {
+  // The condition is true when the "process" module is provided
+  if (process !== global.process) {
+    // prefer local process but global.process has correct "env"
+    process.env = global.process.env
+    global.process = process
+  }
+
+  // to allow building code that import but does not use node.js modules,
+  // webpack will expect this function to exist in global scope
+  Object.defineProperty(globalThis, '__import_unsupported', {
+    value: __import_unsupported,
+    enumerable: false,
+    configurable: false,
+  })
+}
+
+function __import_unsupported(moduleName: string) {
+  const proxy: any = new Proxy(function () {}, {
+    get(_obj, prop) {
+      if (prop === 'then') {
+        return {}
+      }
+      throw new Error(getUnsupportedModuleErrorMessage(moduleName))
```

Comment:
As mentioned in Slack, what do you think about making this a `console.warn()` and then returning `{}` in case the module is imported at the top-level but not actually leveraged?

Reply:
I've tried various things, and discussed it with the edge team.

We believe the current behavior is right: attempt to access a property, function or constructor of an imported node.js module should throw.

Returning an empty object would confuse users because the thrown error is too generic:
```js
import os from 'os'
os.homedir === undefined
os.homedir() // throws TypeError "os.homedir is not a function"
```
Knowing whether "it's not actually leveraged" can not be done at runtime with an empty object. It requires static parsing and instrumentation, which prove to be unreliable in the past, and lead to this PR to change the behavior.

Know, the good news for Vercel Front is that, by simply bumping `@microlink/mql` to the latest and keep using its lightweight version, it won't use clean-stack and the error will be gone! CC @Kikobeats.